src/main/java/org/fenixedu/academic/domain/phd/candidacy/PhdProgramCandidacyProcess.java

```diff
@@ -469,7 +469,7 @@ private void assertCandidacy(final DegreeCurricularPlan dcp, final ExecutionYear
             getCandidacy().setExecutionDegree(executionDegree);
         }
 
-        getCandidacy().setIngression(Ingression.CIA3C);
+        getCandidacy().setIngressionType(IngressionType.findByPredicate(IngressionType::isInternal3rdCycleAccess).orElse(null));
```

Comment:
Couldn't we just link the IngressionType instance CIA3C to some objects the the phd candidacy object graph and just get it from there?